### DISCUSSION TITLE
automatic virtual env activation based on toml specified python version

### DISCUSSION THREAD

**Discussion by farhansaed** [python-poetry/poetry#4066] [Ideas]
My virtual env creation doesn't get triggered automatically as soon I cd into the directory containing toml file. Whereas this happens for pipenv.

Also pipenv integrates with pyenv and if pipfile specified python doesn't existm due to pyenv it autmatically gets installed and virtual env gets activated after installation.

Do we have these two features available in poetry.

**Comment by kozachynskyi** [comment]
1. I'm using https://github.com/darvid/zsh-poetry with zsh and it works. 
2. poetry doesn't call pyenv automatically.

**Reply by farhansaed** [reply]
Thanks for the suggestion. Well it's seems to have maintenance issues. but on other hand a simple script can achieve this so why poetry hasn't done this already. 🤷

**Comment by farhansaed** [comment]
Looking forward for feedback if this is already in the pipeline
